sktime/utils/plotting.py

```diff
@@ -109,3 +109,70 @@ def format_fn(tick_val, tick_pos):
         ax.legend()
 
     return fig, ax
+
+
+def plot_lags(series, lags=1):
+    """Plot one or more lagged versions of a time series
+
+    Parameters
+    ----------
+    series : pd.Series
+        Time series for plotting lags
+    lags : int or array-like
+        Number of lags to plot.
+        int         - plot the specified lag
+        array-like  - plot specified lags in the array/list
+
+    Returns
+    -------
+
+    fig   :  plt.Figure
+    axes  :  plt.Axis or ndarray of plt.Axis objects
+
+    Example
+    -------
+
+    Given the following time series
+
+        >>> np.random.seed(5)
+        >>> x = np.cumsum(np.random.normal(loc=1, scale=5, size=50))
+        >>> s = pd.Series(x)
+
+     If lags is an int, plot a single lagged time series with the
+     specified lag
+
+        >>> plot_lags(s, lags=2) #plot of y(t) with y(t+2)
+
+     If lags is an array-like , plot several lagged time series according to
+     the lags specified in the
+
+        >>> plot_lags(s, lags=[1,2,3]) #plots of y(t) with y(t+1),y(t+2)..
+
+    """
+    _check_soft_dependencies("matplotlib")
+    import pandas as pd
+    import numpy as np
+    import matplotlib.pyplot as plt
+
+    length = 0
+    is_lag_one = isinstance(lags, int)
+
+    if is_lag_one:
+        length = 1
+        lags = [lags]
+    else:
+        length = len(lags)
+
+    fig = plt.figure(figsize=(8, 6 * length))
```

Comment:
Rather iteratively adding the subplots, why not just make them up front (you know how many there will be). 

Also, I think the subplot layout should have multiple columns so the plot does not get too long. 

How about something like:

```python
n_cols = 3
n_rows = int(length / n_cols) + 1
fig, ax = plt.subplots(
    nrows = n_rows, ncols = n_cols, figsize=(..., ....), sharex=True, sharey=True
)
axes = ax.ravel()
```

That would give you the `fig, axes` objects you'll return later and you can just loop over the lags and for each `i, val` do the plotting using something like `pd.plotting.lag_plot(series, lag=val, ax=ax[i])`

Note that I am just using 3 columns as the example, might look fine to have 4 columns in the plotting output.